src/pack.ts

```diff
@@ -89,9 +100,10 @@ export async function pack(this: EsbuildPlugin) {
   const externals = without<string>(this.buildOptions.exclude, this.buildOptions.external);
   const hasExternals = !!externals?.length;
 
+  // get a tree of all production dependencies
   // get a tree of all production dependencies
   const packagerDependenciesList = hasExternals
-    ? await packager.getProdDependencies(this.buildDirPath)
+    ? await (await getPackager('npm')).getProdDependencies(this.buildDirPath)
```

Comment:
@olup do not get it, what if we use yarn?

Reply:
Yarn resolves the deps in a flat way, so I don't get the nesting I need. But this is ugly, I am changing this today.